src/code/z_collision_check.c

```diff
@@ -1,392 +1,3422 @@
+    return 1;
+}
 
-s32 func_8005BA74(UNK_TYPE arg0, UNK_TYPE arg1) {
+// SetAC ColliderJntSphItem
+s32 func_8005BBB0(GlobalContext* globalCtx, ColliderJntSphItem* collider) {
+    func_8005B9E8(globalCtx, &collider->body);
     return 1;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BA84.s")
+// SetOC ColliderJntSphItem
+s32 func_8005BBD4(GlobalContext* globalCtx, ColliderJntSphItem* collider) {
+    func_8005BA1C(globalCtx, &collider->body);
+    return 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BAD8.s")
+// Initialize ColliderJntSph
+s32 func_8005BBF8(GlobalContext* globalCtx, ColliderJntSph* collision) {
+    func_8005B65C(globalCtx, &collision->base);
+    collision->count = 0;
+    collision->list = NULL;
+    return 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BB10.s")
+// Destruct ColliderJntSph (malloc)
+s32 func_8005BC28(GlobalContext* globalCtx, ColliderJntSph* collider) {
+    ColliderJntSphItem* next;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BB48.s")
+    func_8005B6A0(globalCtx, &collider->base);
+    next = collider->list;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BB8C.s")
+    while (next < collider->list + collider->count) {
+        func_8005BB10(globalCtx, next);
+        next++;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BBB0.s")
+    collider->count = 0;
+    if (collider->list != NULL) {
+        ZeldaArena_FreeDebug(collider->list, "../z_collision_check.c", 1393);
+    }
+    collider->list = NULL;
+    return 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BBD4.s")
+// Destruct ColliderJntSph (no malloc)
+s32 func_8005BCC8(GlobalContext* globalCtx, ColliderJntSph* collider) {
+    ColliderJntSphItem* next;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BBF8.s")
+    func_8005B6A0(globalCtx, &collider->base);
+    next = collider->list;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BC28.s")
+    while (next < collider->list + collider->count) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BCC8.s")
+        func_8005BB10(globalCtx, next);
+        next++;
+    }
+    collider->count = 0;
+    collider->list = NULL;
+    return 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005BE50.s")
+// ClObjJntSph
+s32 func_8005BD50(GlobalContext* globalCtx, ColliderJntSph* dest, ColliderJntSphInit_Actor* src) {
+    ColliderJntSphItem* destNext;
+    ColliderJntSphItemInit* srcNext;
+
+    func_8005B6B0(globalCtx, &dest->base, &src->base);
+    dest->count = src->count;
+    dest->list = ZeldaArena_MallocDebug(src->count * sizeof(ColliderJntSphItem), "../z_collision_check.c", 0x5A3);
+
+    if (dest->list == NULL) {
+        dest->count = 0;
+        osSyncPrintf("\x1b[31m");
```

Comment:
Use VT macros.
see vt.h
```suggestion
        osSyncPrintf(VT_FGCOL(RED));
```